.github/check-eof-newline.sh

```diff
@@ -1,5 +1,6 @@
 #!/bin/sh
 # Checks that all text files end with a newline.
+echo "Checking for newline at EOF..."
```

Comment:
Suggest extracting these changes to their own commit/issue/PR as they are technically not related to the issue at hand.

Reply:
After discussion, changes can be left in this PR, but should be separated into different commits so that they will show up in the squashed commit upon merge.